source/Ship.cpp

```diff
@@ -2634,7 +2707,147 @@ bool Ship::IsDisabled() const
 
 	double minimumHull = MinimumHull();
 	bool needsCrew = RequiredCrew() != 0;
-	return (hull < minimumHull || (!crew && needsCrew));
+	bool outOfEnergy = IsOutOfEnergy();
+	return hull < minimumHull || (!crew && needsCrew) || outOfEnergy;
+}
+
+
+
+// Determine if the ship has any usable weapons.
+bool Ship::IsArmed() const
+{
+	return IsArmed(false);
```

Comment:
A ship can be completely unarmed (no antimissile and no offensive weapons) or defensively armed (only antimissile).  Different parts of the AI care about this differently from the callers so it felt okay to overload in this case because that's simpler I think.  The decision needs to be made on call so I don't think this is something that can go into the header file maybe?